components/switch/src/Switch.js

```diff
@@ -62,14 +48,143 @@ const asArray = val => (Array.isArray(val) ? val : [val]);
 // ==============================
 // Component
 // ==============================
+export const Switch = ({
+	name,
+	label,
+	checked: isChecked,
+	onChange,
+	size,
+	block,
+	flipped,
+	toggleText,
+	disabled,
+	srOnlyText,
+}) => {
+	const { COLORS, [pkg.name]: localBrandTokens } = useBrand();
+	const mq = useMediaQuery();
+	const [checked, setChecked] = useState(isChecked);
+	const [switchId] = useState(`switch-${shortid.generate()}`);
+
+	const sizeArr = responsiveMap(asArray(size));
+
+	const localTokens = {
+		labelCss: {},
+		textCss: {},
+		toggleCss: {},
+		toggleTextCss: {},
+	};
+
+	merge(localTokens, localBrandTokens);
 
-export const Switch = ({ size, ...props }) => {
-	const flexiSize = responsiveMap(asArray(size));
+	useEffect(() => {
+		setChecked(isChecked);
+	}, [isChecked]);
+
+	const handleChange = name => () => {
+		if (onChange) {
+			onChange(name);
+		} else {
+			setChecked(!checked);
+		}
+	};
 
 	return (
-		<SwitchContext.Provider value={{ flexiSize }}>
-			<SwitchLabel {...props} />
-		</SwitchContext.Provider>
+		<label
+			htmlFor={switchId}
```

Comment:
You don't need to use and id or `for` attribute if you wrap the input inside a label (which you're doing here) and avoid using unique IDs all-together.